pkg/performanceprofile/controller/performanceprofile_controller.go

```diff
@@ -214,6 +274,44 @@ func (r *PerformanceProfileReconciler) getInfraPartitioningMode() (pinning apico
 	return infra.Status.CPUPartitioning, nil
 }
 
+func (r *PerformanceProfileReconciler) getContainerRuntimeName(profile *performancev2.PerformanceProfile) (mcov1.ContainerRuntimeDefaultRuntime, error) {
+	ctrcfgList := &mcov1.ContainerRuntimeConfigList{}
+	if err := r.List(context.Background(), ctrcfgList); err != nil {
+		if k8serros.IsNotFound(err) {
+			return mcov1.ContainerRuntimeDefaultRuntimeRunc, nil
+		}
+		return "", err
+	}
+
+	profileMCPSelector := labels.Set(profile.Spec.MachineConfigPoolSelector)
```

Comment:
I do not think this is enough. Our mcpSelector is an optional field and we actually have some logic to detect master /  worker pools too.

See https://github.com/openshift/cluster-node-tuning-operator/blob/master/pkg/performanceprofile/controller/performanceprofile/components/profile/profile.go#L11 and the newer https://github.com/openshift/cluster-node-tuning-operator/blob/master/pkg/performanceprofile/controller/performanceprofile_controller.go#L299

Reply:
Yes, we need to review the matching logic and very likely to reuse it to make things consistent

Reply:
The `ContainerRuntimeCofing` object only has a `machineConfigPoolSelector`.
We can detect the MCP by the profile, and then look for labels in the MCP that are matching the `machineConfigPoolSelector`:

```
mcp, err := r.getMachineConfigPoolByProfile(profile)
...

//ctrcfg is mcov1.ContainerRuntimeConfig object
ctrcfgSelector, err := metav1.LabelSelectorAsSelector(ctrcfg.Spec.MachineConfigPoolSelector)
...

if ctrcfgSelector.Matches(labels.Set(mcp.Labels)) {
// we found the ContainerRuntimeConfig that is associate with the profile
}

Is it better now?

Reply:
Yeah, this looks better I think